cmd/apiserver/app/server/run_server.go

```diff
@@ -42,7 +42,8 @@ func RunServer(opts *ServiceCatalogServerOptions) error {
 func runTPRServer(opts *ServiceCatalogServerOptions) error {
 	tprOpts := opts.TPROptions
 	glog.Infoln("Installing TPR types to the cluster")
-	if err := tpr.InstallTypes(tprOpts.clIface); err != nil {
+	tprInstaller := tpr.NewInstallerTPRs(tprOpts.clIface)
```

Comment:
@simonleung8 adding another code generator as well as a bunch of new generated code is greatly impactful. And since it's for testing purposes, I don't believe that it's worth the impact.

What I'm suggesting is that you reduce the interface surface by a lot -- which is possible in this case -- and you pass to the install function, and then just mock that (much smaller) interface. There is nothing wrong with hand-coding a small test implementation to the TPR interface; of all the code generators, your fingers and mind are the best proven tool we have.

Regarding the `Installer` type - can you explain how it makes dependency injection easier? The reason I ask is that it looks like you put the `clientset.Interface` inside the struct instead of passing it as a function parameter, and I don't understand why the former is better than the latter.